packages/framework/aqueduct/src/data-objects/dataObject.ts

```diff
@@ -32,6 +36,17 @@ export abstract class DataObject<O extends IFluidObject = object, S = undefined,
     private internalRoot: ISharedDirectory | undefined;
     private readonly rootDirectoryId = "root";
 
+    /**
+     * Unique ID of this DataObject.
+     * This id is a random guid that is created when data store is initially created
+     * It is preserved in container and can be accessed next time this data object is loaded.
+     */
+    public get uniqueId(): string {
```

Comment:
I don't think we should add anything here, esp. without an example of good usage.  This pattern is easy enough to implement from the outside without integrating it into the base DO class.